packages/server/dataloader/customLoaderMakers.ts

```diff
@@ -634,3 +636,78 @@ export const billingLeadersIdsByOrgId = (parent: RootDataLoader) => {
     }
   )
 }
+
+export const saml = (parent: RootDataLoader) => {
+  return new NullableDataLoader<string, SAMLSource | null, string>(
+    async (keys) => {
+      const pg = await getKysely()
+      const res = await Promise.all(
+        keys.map(async (id) => {
+          return pg
+            .selectFrom('SAMLDomain')
+            .innerJoin('SAML', 'SAML.id', 'SAMLDomain.samlId')
+            .where('SAML.id', '=', id)
+            .groupBy('SAML.id')
+            .selectAll('SAML')
+            .select(({fn}) => [fn.agg<string[]>('array_agg', ['SAMLDomain.domain']).as('domains')])
+            .limit(1)
+            .executeTakeFirst()
+        })
```

Comment:
yep, i was on the fence about this! We sacrifice a little readability for a more performant query, and I figured that was fine for something like SAML since the odds of a dataloader ever handling more than 1 or 2 is pretty slim. that said, I didn't say this in a comment & if someone cites my code as an example, i'd be sad if they used it for something more common like a Task. I'll go ahead & make it more performant so we can use it as an example.